Mathlib/Data/Matrix/Rank.lean

```diff
@@ -274,4 +275,34 @@ theorem rank_eq_finrank_span_row [LinearOrderedField R] [Finite m] (A : Matrix m
   rw [← rank_transpose, rank_eq_finrank_span_cols, transpose_transpose]
 #align matrix.rank_eq_finrank_span_row Matrix.rank_eq_finrank_span_row
 
+section MulNonsingInv
+
+/-! ### Lemmas about pre or post multiplying by an invertible matrix. -/
+
+/-- Pre (Left) multiplying by an invertible matrix does not change the rank -/
+lemma rank_mul_isUnit [DecidableEq n] [CommRing R]
+    (A: Matrix n n R) (B: Matrix m n R) (hA: IsUnit A.det) :
+    (B ⬝ A).rank = B.rank := by
+  rw [Matrix.rank, mulVecLin_mul, LinearMap.range_comp_of_range_eq_top, ←Matrix.rank ]
+  rw [LinearMap.range_eq_top] -- On separate line since implicit rfl works in prev line
+  intro x
+  use ((A⁻¹).mulVecLin x)
+  rwa [mulVecLin_apply, mulVecLin_apply, mulVec_mulVec, mul_nonsing_inv, one_mulVec]
```

Comment:
```suggestion
  · rw [LinearMap.range_eq_top] 
    intro x
    use ((A⁻¹).mulVecLin x)
    rwa [mulVecLin_apply, mulVecLin_apply, mulVec_mulVec, mul_nonsing_inv, one_mulVec]
```
The comment as is is rather confusing (I didn't understand it until opening this in VSCode). The point is that the previous rewrite closed the goal, and you are now working on a side goal. This suggestion is perhaps clearer, but a comment explicitly mentioning that the rest of the proof is the side goal generated by `range_comp_of_range_eq_top` would be better.

Or even better --- prove the side goal *first*, in a `have`!